network-rfc/12_contracts_design.md

```diff
@@ -0,0 +1,90 @@
+Once deployed, SQD token providers can stake their tokens directly into the PortalProxy by calling the stake function with the portal address and desired amount. 
+
+During this collection phase, the portal remains in a "Collecting" state where it accumulates SQD deposits from multiple providers until either the target amount is reached or the deposit deadline passes. 
+If the target is met before the deadline, the data consumer can trigger the activate function to transition the portal to its active distribution phase.
+
+However, if the deadline expires without reaching the target, the portal is marked as failed, triggering a full refund of both the consumer's budget and all staked SQD tokens back to their respective owners.
+
+### Active Distribution and Fee Routing
+Once activated, the portal enters its Active state where it begins distributing 
+
+Throughout this active period, the data consumer can call the distribute function to inject tokens into the contract, which will be distributed across SQD providers etc. 
+This amount is  distributed based on the FeeRouterModule, a separate admin-controlled contract responsible for splitting the fees according to configurable basis point allocations (defaulting to 50% for SQD providers, 45% for the worker reward pool, and 5% for burning).
+
+The FeeRouterModule holds the actual BPS.
+ During both the staking and distribution phases, the system can trigger external Hooks at key moments (before and after staking, distribution, and exits), allowing for customized behavior such as  additional protocol token rewards layered on top of base distributions etc.
+ Similar to UniswapV4 Hooks.
+
+### Reward Claims, Exits, and Closure
+
+While the portal is active, SQD providers can claim their proportional share of accumulated rewards at any time by calling the claimRewards function on the PortalProxy, which calculates their share based on their staked balance relative to the total tokens in the portal and transfers the corresponding tokens to them. The portal continues distributing as long as the data consumer injects tokens through the distribute function, with all distributions based on the FeeRouterModule configured splits.
+
+When SQD providers stake their tokens into the portal, they lock them for a minimum duration period. 
+After this minimum lock period expires, providers can request to exit the portal by calling requestExit with their desired withdrawal amount. 
+However, exits are subject to a time-delay mechanism designed to prevent sudden liquidity shocks: 
+the exit delay consists of a base period of 1 epoch plus a percentual delay calculated by the amount being withdrawn. 
+The system allows a maximum of 1% of the total portal liquidity to exit per epoch, meaning if a provider wants to exit 5% of the liquidity, they must wait 1 epoch (base) plus 5 additional epochs (one epoch per 1% of liquidity), totaling 6 epochs before their full withdrawal is processed. 
```

Comment:
Should it be 1% of the total pool or fixed amount in SQD? I think we agreed on a number 100k SQD per epoch on the meeting